unit-testing/presentation.html

```diff
@@ -2,43 +2,512 @@
+                different data every time and etc.
+              </li>
+            </ul>
+            <pre
+              class="fragment"
+            ><code data-trim data-noescape data-line-numbers>
+          // Mock functions​​
+          jest.mock(moduleName, factory)​​
+          jest.unmock(moduleName)​​
+          jest.requireActual(moduleName)​​
+          jest.resetModules()​​
+          jest.isolateModules(fn)​​
+          // Mock modules​​
+          jest.fn(implementation?)​​
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+          </section>
+        </section>
+        <!-- 19 slide -->
+        <section class="text-and-conding-slide">
+          <h4 class="slide-name">Snapshots​​</h4>
+          <ul>
+            <li class="fragment">
+              Snapshot tests are used when you want to make sure your UI does
+              not change unexpectedly.
+            </li>
+            <li class="fragment">
+              The test will fail if the two snapshots do not match.
+            </li>
+            <li class="fragment">
+              However snapshots themselves do not ensure that your component
+              render logic is correct.
+            </li>
+            <li class="fragment">
+              Use small snapshots because large snapshots can quickly become
+              hard -> keep no-large-snapshots rule.
+            </li>
+          </ul>
+          <pre class="fragment"><code data-trim data-line-numbers>
+          it('should match snapshot', () => {
+            const {toJSON} = render(<mybutton title="title"></mybutton>);
+            expect(toJSON()).toMatchSnapshot();
+          });</code></pre>
+        </section>
+        <!-- 20 slide -->
+        <section>
+          <!-- 20/0 slide -->
+          <section class="bullets-slide">
+            <h4 class="slide-name">Test coverage​</h4>
+            <ul>
+              <li class="fragment">
+                With that being said it is generally accepted that 80% coverage
+                is a good goal to aim for.
+              </li>
+              <li class="fragment">
+                Trying to reach a higher coverage might turn out to be costly,
+                while not necessary producing enough benefit.​
+              </li>
+              <li class="fragment">
+                100% unit test coverage does not mean we had good tests, or even
+                that the tests are complete.
+              </li>
+              <li class="fragment">
+                The tests could be missing important data and only testing with
+                data that succeeds, failing to test data that causes failures.
+              </li>
+              <li class="fragment">
+                Merely executing the code without checking its functionality
+                still counts in the coverage metrics.
```

Comment:
This slide looks like an article.
It's better to move most the most text into the notes and leave a simple list like
```
- 80% coverage is a good goal;
- higher coverage might be costly and doesn't give enough benefit
- 100% coverage doesn't mean good tests or that tests are complete
- (I don't quite understood it but it must be something like: "tests could be missing important data and test with data that succeeds and failing to test data that causes failures". What did you mean?)
- coverage metrics includes even the code that isn't checked for functionality
```